### DISCUSSION TITLE
why dont post brew builds publicly?

### DISCUSSION THREAD

**Discussion by DeveloperKev** [Homebrew/discussions#4012] [Tap maintenance and Homebrew development]
Hi
i respect the good work, but why doesnt brew publish the builds directly? what is keeping brew from doing that?
Also, previously i got hands with some malicious libraries using brew. theres no password protection of brew, anyone could make a .rb and execute. this is real.

**Comment by Bo98** [comment]
> why doesnt brew publish the builds directly? what is keeping brew from doing that?

Can you elaborate what you mean by publishing "directly"?

> Also, previously i got hands with some malicious libraries using brew. theres no password protection of brew, anyone could make a .rb and execute. this is real.

As in someone compromises your local system and makes a .rb file?

The remote copy homebrew-core is protected by GitHub access control. For third party taps, it's very much a case of only use those from trusted people (as is the case for any software).

**Comment by DeveloperKev** [comment]
public binaries to be specific*

some third party codes are visible in brew repo.

**Reply by p-linnane** [reply]
Are you referring to bottles? You can browse the packages and pull them down here if you want: https://github.com/orgs/Homebrew/packages. You can also run `brew fetch <formula>` to pull down the tarballs as well.

You're responsible for what you install. You need to make sure you trust what you are installing, it is not up to Homebrew to protect you from that. If you find something that is malicious, feel free to report it and we will investigate to determine if it should be removed. If you're talking about your local system like @Bo98 said, that's completely out of scope for what we can do to help you.

**Reply by DeveloperKev** [reply]
> Are you referring to bottles? You can browse the packages and pull them down here if you want: https://github.com/orgs/Homebrew/packages. You can also run `brew fetch <formula>` to pull down the tarballs as well.
> 
> You're responsible for what you install. You need to make sure you trust what you are installing, it is not up to Homebrew to protect you from that. If you find something that is malicious, feel free to report it and we will investigate to determine if it should be removed. If you're talking about your local system like @Bo98 said, that's completely out of scope for what we can do to help you.

docker pull ?

**Reply by SMillerDev** [reply]
Brew doesn't build containers so no `docker pull`

**Reply by DeveloperKev** [reply]
i think, u've told me before that brew stores binaries in a non-usable format, right?
sorry, if i'm wrong, but as an open source, brew should provide tarballs.

**Reply by p-linnane** [reply]
As I said above, you can get the tarballs by running `brew fetch <formula>`.

**Reply by DeveloperKev** [reply]
> As I said above, you can get the tarballs by running `brew fetch <formula>`.

yeah, i just tested.
can u confirm one thing: for gcc compiler, is it built so the dependencies will work, remaining in the same folder?

**Reply by SMillerDev** [reply]
>  but as an open source, brew should provide tarballs.

It really doesn't have to, just like the Linux kernel doesn't have to make tarballs. It just has to make the source available. It could be distributed by trained pigeons if they wanted to.

**Reply by DeveloperKev** [reply]
> gcc compiler, is it built so that dependencies will only work placing the files in the same folder?

@SMillerDev  anything on this?

**Reply by SMillerDev** [reply]
No, I don't even think there is a way to make it do that. Here are the docs on how GCC works https://gcc.gnu.org/onlinedocs/

**Reply by DeveloperKev** [reply]
> No, I don't even think there is a way to make it do that. Here are the docs on how GCC works https://gcc.gnu.org/onlinedocs/

gcc should be build after placing the dependencies in a certain location to be called from. that's why, i am asking if all of the dependencies along with gcc are stored in `Cellar` folder.

**Reply by DeveloperKev** [reply]
@SMillerDev gcc will look for dependencies, i think u'll understand by now.

**Reply by SMillerDev** [reply]
Well, if you use brew fetch to download Homebrew binaries it's up to you to recreate the Homebrew dependency tree in the locations it needs them in if you want to use those binaries since that's not a supported use of Homebrew software.

**Reply by SMillerDev** [reply]
If you want to know where they need to be stored you need to read the formulae files and possibly also inspect the linkage of the binaries.
